frontend/src/container/GridTableComponent/index.tsx

```diff
@@ -25,28 +27,28 @@ function GridTableComponent({
 	...props
 }: GridTableComponentProps): JSX.Element {
 	const { t } = useTranslation(['valueGraph']);
+
+	const { currentQuery } = useQueryBuilder();
+
+	// create columns and dataSource in the ui friendly structure
 	const { columns, dataSource: originalDataSource } = useMemo(
 		() =>
-			createTableColumnsFromQuery({
-				query,
-				queryTableData: data,
-			}),
-		[data, query],
+			createColumnsAndDataSource((data as unknown) as TableData, currentQuery),
+		[currentQuery, data],
 	);
+
 	const createDataInCorrectFormat = useCallback(
 		(dataSource: RowData[]): RowData[] =>
 			dataSource.map((d) => {
 				const finalObject = {};
-				const keys = Object.keys(d);
-				keys.forEach((k) => {
-					const label = get(
-						columns.find((c) => get(c, 'dataIndex', '') === k) || {},
-						'title',
-						'',
+
+				// we use the order of the columns here to have similar download as the user view
+				columns.forEach((k) => {
+					set(
+						finalObject,
+						get(k, 'title', '') as string,
+						get(d, get(k, 'dataIndex', ''), 'n/a'),
```

Comment:
n/a ?

Reply:
yes if the value of that particular column is not present we input `n/a` similar to the logic in api response. ideally we won't enter here since all the rows are properly filled from the response but just added a fallback in case